File: src/main/java/com/examportal/dto/UserSignupDTO.java
package com.examportal.dto;

import com.examportal.entities.UserRole;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
public class UserSignupDTO {
    
    @NotBlank(message = "Username must be provided")
    @Size(min = 3, max = 50, message = "Username must be between 3 and 50 characters")
    private String username;
    
    @NotBlank(message = "Email must be provided")
    @Email(message = "Invalid email format")
    private String email;
    
    @NotBlank(message = "Password must be provided")
    @Pattern(regexp = "((?=.*\\d)(?=.*[a-z])(?=.*[#@$*]).{5,20})", 
             message = "Password must contain at least one digit, one lowercase letter, one special character (#@$*), and be 5-20 characters long")
    private String password;
    
    @NotNull(message = "Role must be provided")
    private UserRole role;
}

File: Dotnet3/ExamPortal/DTOs/UserSignupDTO.cs
using ExamPortal.Models;
using System.ComponentModel.DataAnnotations;

namespace ExamPortal.DTOs
{
    public class UserSignupDTO
    {
        [Required(ErrorMessage = "Username must be provided")]
        [StringLength(50, MinimumLength = 3, ErrorMessage = "Username must be between 3 and 50 characters")]
        public string Username { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "Email must be provided")]
        [EmailAddress(ErrorMessage = "Invalid email format")]
        public string Email { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "Password must be provided")]
        [StringLength(100, MinimumLength = 6, ErrorMessage = "Password must be at least 6 characters long")]
        public string Password { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "First name must be provided")]
        [StringLength(50, MinimumLength = 2, ErrorMessage = "First name must be between 2 and 50 characters")]
        public string FirstName { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "Last name must be provided")]
        [StringLength(50, MinimumLength = 2, ErrorMessage = "Last name must be between 2 and 50 characters")]
        public string LastName { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "Role must be provided")]
        [RegularExpression("^(STUDENT|TEACHER)$", ErrorMessage = "Role must be either STUDENT or TEACHER")]
        public string Role { get; set; } = string.Empty;
    }
}
